### PR TITLE
Move app menu to vue

### PR DESCRIPTION
**Pull request by juliusknorr**
Move app menu to vue in preparation for further adjustments of #33568

- [x] Fix current app indicator
- [x] Properly fix resizing
- [x] Notification indicator in actions menu
- [x] Aria-label on more apps button
- [x] Move to bar active indicator as discussed with @jancborchardt 
- [x] Fix focus frame

https://user-images.githubusercontent.com/3404133/187261115-4393d2d6-b6bf-484d-aba5-ca451a753e81.mov

### CODE REVIEW

File: core/src/components/AppMenu.vue [hunk 1]
```diff
@@ -106,6 +106,7 @@ $header-icon-size: 20px;
 .app-menu {
 	width: 100%;
 	display: flex;
+	overflow: hidden;
```
Comment (juliushaertl) [inline core/src/components/AppMenu.vue:109]: This cuts off the app title at the first/last item

**Comment by CarlSchwan** [comment]
The focus frame is now touching the border of the header, causing some contrast issue since there is no clear separation between the focus frame and outside the focus frame.

image

**Comment by szaimen** [comment]
@juliushaertl would it be possible to add an API for re-ordering the apps or hiding some of them? I am thinking about the apporder app that will probably break after this PR...

**Comment by juliushaertl** [comment]
> @juliushaertl would it be possible to add an API for re-ordering the apps or hiding some of them? I am thinking about the apporder app that will probably break after this PR...

Possible yes, but not planned yet. There is some long discussion about this in https://github.com/nextcloud/server/issues/4917

**Comment by juliushaertl** [comment]
@CarlSchwan Addressed :)

File: core/src/components/AppMenu.vue [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+			return this.appList.slice(this.appLimit)
+		},
+		appLabel() {
+			return (app) => app.name
+				+ (app.active ? ' (' + t('core', 'Currently open') + ')' : '')
+				+ (app.unread > 0 ? ' (' + n('core', '{count} notification', '{count} notifications', app.unread, { count: app.unread }) + ')' : '')
+		},
+	},
+	mounted() {
+		window.addEventListener('resize', this.resize)
```
Comment (artonge) [inline core/src/components/AppMenu.vue:90]: FYI, there is a "new" [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) interface to detect size change on a specific object.

But `addEventListener` works in this case, so feel free to ignore.

Comment (juliushaertl) [reply core/src/components/AppMenu.vue]: Switched over as that may cover cases where the container gets resized during loading if not all right side menu entries are rendered in time (even through not likely).

**Review by artonge — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Comment by PVince81** [comment]
there are JS test failures related to snap/snapper that need to be adjusted
maybe we changed the width?

**Comment by juliushaertl** [comment]
Thanks @danxuliu ❤️

**Review by jancborchardt — CHANGES_REQUESTED** [review]
Some details:
- [x] The active indicator is a bit low down, would be nice if it’s visually inbetween the icon and the content. Just a few px further up :)
- [ ] With a light background selected from the Dashboard, is the same logic of making the header bar icons dark, so they are visible?
- [ ] As per the video, the focus effect is still overlapping the text, or is the vid outdated, as in @CarlSchwan’s comment it also looked different?

**Comment by juliushaertl** [comment]
>  With a light background selected from the Dashboard, is the same logic of making the header bar icons dark, so they are visible?

Something we can do once the background image is a global option: https://github.com/nextcloud/server/pull/33733

> As per the video, the focus effect is still overlapping the text, or is the vid outdated, as in @CarlSchwan’s comment it also looked different?

I didn't touch that yet, to first get the major change in, still tracked as open in https://github.com/nextcloud/server/issues/33741

> The active indicator is a bit low down, would be nice if it’s visually inbetween the icon and the content. Just a few px further up :)

Will adjust.

**Comment by juliushaertl** [comment]
Screenshot 2022-08-31 at 14 39 17

@jancborchardt

File: core/src/init.js [hunk 3]
```diff
@@ -36,7 +36,8 @@ import { setUp as setUpMainMenu } from './components/MainMenu'
 import { setUp as setUpUserMenu } from './components/UserMenu'
 import PasswordConfirmation from './OC/password-confirmation'
 
-const breakpointMobileWidth = getComputedStyle(document.documentElement).getPropertyValue('--breakpoint-mobile')
+// keep in sync with core/css/variables.scss
+const breakpointMobileWidth = 1024
```
Comment (szaimen) [inline core/src/init.js:40]: not sure about this... yet another place where we need to maintain this number...

**Review by jancborchardt — APPROVED** [review]
Looks super nice! :)

**Review by CarlSchwan — APPROVED** [review]

**Comment by juliushaertl** [comment]
Failure unrelated

**Pull request merged**